compiler/rustc_smir/src/rustc_internal/pretty.rs

```diff
@@ -0,0 +1,291 @@
+use std::io;
+
+use rustc_middle::ty::TyCtxt;
+use stable_mir::{
+    mir::{Mutability, Operand, Rvalue, StatementKind},
+    ty::{RigidTy, TyKind},
+    CrateItem,
+};
+
+use super::{internal, run};
+
+pub fn write_smir_pretty<'tcx>(tcx: TyCtxt<'tcx>, w: &mut dyn io::Write) -> io::Result<()> {
+    writeln!(w, "// WARNING: This is highly experimental output it's intended for stable-mir developers only.").unwrap();
+    writeln!(w, "// If you find a bug or want to improve the output open a issue at https://github.com/rust-lang/project-stable-mir.").unwrap();
+    run(tcx, || {
+        let items = stable_mir::all_local_items();
+        items.iter().for_each(|item| {
+            // Because we can't return a Result from a closure, we have to unwrap here.
```

Comment:
you can use `map` instead of `for_each` and return a Result instead.

Reply:
Yeah makes sense!